src/shelter/use-shelter.hook.ts

```diff
@@ -41,6 +45,25 @@ export const useShelter = () => {
           }
         }),
 
+      send$
+        .pipe(
+          switchMap(data =>
+            Shelter.revealSecretKey$(data.from).pipe(
+              map((privateKey): [string | undefined, SendInterface] => [privateKey, data])
+            )
+          ),
+          withLatestFrom(tezos$),
+          switchMap(([[privateKey, data], tezos]) => {
+            if (privateKey) {
+              tezos.setProvider({
+                signer: new InMemorySigner(privateKey)
+              });
+            }
+            return tezos.contract.transfer({ to: data.to, amount: parseInt(data.amount, 2) });
```

Comment:
You will still make a request, even if there is no `privateKey`.  
Add value check before `map`
```
switchMap(data =>
            Shelter.revealSecretKey$(data.from).pipe(
              switchMap(value => (value === undefined ? throwError(`Failed to reveal private key`) : of(value))),
              map((privateKey): [string, SendInterface] => [privateKey, data])
            )
          ),
```